src/renderers/dom/client/ReactDOMTreeTraversal.js

```diff
@@ -80,13 +80,18 @@ function getParentInstance(inst) {
 }
 
 /**
- * Simulates the traversal of a two-phase, capture/bubble event dispatch.
+ * Simulates the traversal of a two-phase, capture/bubble event dispatch
+ * with skipping disabled instances.
  */
 function traverseTwoPhase(inst, fn, arg) {
   var path = [];
   while (inst) {
     path.push(inst);
-    inst = inst._hostParent;
+    var parent = getParentInstance(inst);
+    while (parent && parent._currentElement && parent._currentElement.props.disabled) {
```

Comment:
I'm pretty sure this isn't safe - `disabled` only has meaning on specific elements. eg `<div disabled><button onClick=... /></div>` should still fire as the button is not disabled in the DOM.

This is also going to be super expensive. I don't think we can do this like this, even if we managed to just stop at form boundaries.